rustls/src/crypto/aws_lc_rs/ticketer.rs

```diff
@@ -38,24 +38,6 @@ impl Ticketer {
             make_ticket_generator,
         )?))
     }
-
-    /// Make the recommended `Ticketer`.  This produces tickets
-    /// with a 12 hour life and randomly generated keys.
-    ///
-    /// The `Ticketer` uses the [RFC 5077 §4] "Recommended Ticket Construction",
-    /// using AES 256 for encryption and HMAC-SHA256 for ciphertext authentication.
-    ///
-    /// [RFC 5077 §4]: https://www.rfc-editor.org/rfc/rfc5077#section-4
-    #[cfg(not(feature = "std"))]
-    pub fn new<M: crate::lock::MakeMutex>(
```

Comment:
This seems like a kind of big semver compatibility issue? Unless we have a way of finding everyone using rustls in no_std context (I guess releasing this is potentially a decent way of getting them into our issue tracker), I would be hesitant to release this as 0.23.x.

Reply:
I guess the other option is to reinstate this, and keep around the previous version of the ticketswitcher until the next breaking release? But I don't know where that leaves existing users (who already have a breaking API change if something in their dependency tree turns on `std`).

I think in the next breaking release we should do a pass to improve this pattern, so it looks like:

lock.rs:
```rust
struct StdMutex;
impl MakeMutex for StdMutex { ... obvious stuff ... }
```

elsewhere:
```rust
#[cfg(feature = "std")]
pub fn new(...) { new_detail::<StdMutex>() }

pub fn new_detail<M: crate::lock::MakeMutex>(...) { }
```

(this deals with that kind of feature-variant breaking change, but does not deal with the issue that `new_detail` leaks the otherwise-abstract details -- like "this uses a lock" -- into the public API.)